Studentmediene/kapina-frontend
# Refactor/67 upgrade react router

Well... this was a larger task than i expected.

This PR upgrades `react-router-dom` and related dependencies to the newest versions. 
The browser's navigation history is now synced with Redux, and enables programmatic navigation through the use of `store.dispatch(push(<path>))`. Code splitting and hot reloading is still enabled through webpack, where all `System.import` statements in `routes.js` generates an independent bundle. From now on, all of Redux' `connect` statements have to be wrapped in `withRouter` to enable the UI to be updated with location updates. 

## Work that should be done in a later PR:
- Use nesting of routes instead of declaring them all in a single file.
- Refactor the `App` component so that it does not have to receive all routes as props

## How to test: 
Install new dependencies, start devServer and navigate through all pages of the website. Remember to check an invalid URL (404).
